openmdao/core/group.py

```diff
@@ -200,6 +200,8 @@ class Group(System):
         within this group, keyed by active response.  These determine if contributions
         from all ranks will be added together to get the correct input values when derivatives
         in the larger model are being solved using reverse mode.
+    _bad_conn_vars : set
+        Set of variables involved in invalid connections.
```

Comment:
Set of tuples (input_var, output_var) representing an invalid connection

Reply:
It's just a set of variable names, not tuples.  I just pass a tuple containing both the input and output name to the `update` method to avoid having to make 2 calls to the `add` method.